src/library/Api/API.js

```diff
@@ -188,37 +202,33 @@ const API = {
      * Make a request to the API
      * @param {string} method The HTTP method to use
      * @param {string} url The URL to call
-     * @param {object} [params] The parameters to send
+     * @param {object|string} [params] The parameters to send
      * @param {function} [successHandler] The function to call if the request is successful
      * @param {function} [errorHandler] The function to call if the request is unsuccessful
      *
      * @documentation https://fossbilling.org/docs/api/javascript
      */
     makeRequest: function (method, url, params, successHandler, errorHandler) {
-        // If the parameters are not set, set them to an empty object
-        params = (params) ? params : {};
-
-        // If the request didn't specify a CSRF token, use the one set in the page headers
-        params.CSRFToken = (params.CSRFToken) ? params.CSRFToken : Tools.getCSRFToken();
-
         url = new URL(url);
-
-        var body = params;
-
+        if(typeof params === 'object'){
+          if(!params.CSRFToken){ params.CSRFToken=Tools.getCSRFToken()}
+        }
         // Loop through the parameters and add them to the URL as a query string
         // GET requests should have their parameters in the query string and POST requests should have them in the body
         if (method.toLowerCase() === "get") {
-            var getParams = params;
-            // Convert JSON strings to an Object
-            if (Tools.isJSON(params)) {
-                getParams = JSON.parse(params);
+          if(typeof params === 'object'){
+            Object.keys(params).forEach(key => url.searchParams.append(key, params[key]));
+          }else{
+            if(params){
```

Comment:
In this case, is there any way we can add the token? (as long as it isn't going to potentially cause issues or cause a significant amount of complication)

Also, is there any chance that any of these changes could cause issues with the client theme since that still uses jquery?

Reply:
Client theme have still their own api

https://github.com/FOSSBilling/FOSSBilling/blob/7ef26a2888b5cb4225e37066c990c08c3875de05/src/themes/huraga/assets/js/bb-jquery.js#L34-L38

https://github.com/FOSSBilling/FOSSBilling/blob/7ef26a2888b5cb4225e37066c990c08c3875de05/src/themes/huraga/assets/js/bb-jquery.js#L9

Reply:
Ah, for some reason I thought it processed the object data through jquery and then sent it through the API wrapper.

Reply:
No there are still some old api class in jQuery in the admin part they should still work